### ISSUE TITLE
Remote SSH Connection Crashed after openning a Jupyter Notebook on the remote server

### ISSUE THREAD

**Issue by ductai199x** [microsoft/vscode#195617]
Does this issue occur when all extensions are disabled?: No

- VS Code Version: 1.83.1
- OS Version: Windows 11 latest

Steps to Reproduce:

1. Open up a remote ssh connection to a remote server
2. Open a Jupyter Notebook stored on that remote server. It can be a brand new Notebook or something already existed.
3. VSCode crashed with "Cannot reconnect. Please reload the window."

It could be a problem with VSCode, or the Python / Jupyter extension. This did not happen with VSCode 1.83.0 so something in the update could have messed things up.

**Comment by nathanmargaglio** [comment]
Same issue here.  I've found that some workspaces (such as selecting the parent directory) appears to avoid the issue.  The "Cannot reconnect.  Please reload the window." error only occurs when you actually select a kernel.  I don't have any .vscode configuration in either of these workspaces, so I'm not really sure what the issue is.

**Comment by vishu-tyagi** [comment]
After downgrading to VSCode 1.83.0, I'm still facing the same issue

**Comment by ductai199x** [comment]
> After downgrading to VSCode 1.83.0, I'm still facing the same issue

You need to first remove the .vscode_server folder on the remote server, then reinstall vscode on your local computer with the older version, and then reinstall the extensions on remote server (Python, Jupyter, etc.). The problems go away after I did these steps

**Comment by edsealing** [comment]
I'm seeing the same problem when connecting to Remote-SSH as well as WSL. occurs in both "code" and "code-insiders" (from MS App Store). Specifically occurs for me when attempting to "Select Kernel".

**Comment by live4fun** [comment]
same problem

**Comment by watchwater** [comment]
I just encountered the same issue. When I was selecting the Jupyter kernel, the SSH connection began to disconnect. At the same time, I noticed through the 'htop' command in the terminal that the CPU usage began to spike significantly as the SSH connection was interrupted.

**Comment by tubagundem** [comment]
I have the same problem since yesterday. I can work on other files via vscode ssh connection. However, when I open any Jupyter notebook, it asks for the password again and the connection times out.

**Comment by jack-tobin** [comment]
Same issue here, specifically when saving-as a new notebook created with the "Create: New Jupyter Notebook" shortcut. Breaks my connection to WSL in every open window. Only quick fix I've found is to restart VSCode entirely

**Comment by AndrewGrishin** [comment]
Have the same problem, but neither reloading, nor reinstall made any impact.

**Comment by keitarobr** [comment]
Same issue here - can't use at all.

**Comment by federicoromeo** [comment]
same problem here

**Comment by AndrewGrishin** [comment]
Discovered the solution that worked for me (**local**: macOS Ventura 13.6, M1, **remote server**: Ubuntu 22.04.3 LTS x86_64).

1. Backup `settings.json` (if you've made any local configurations).
2. Close VS Code locally and remotely.
3. Completely delete VS Code from your local machine via steps in #[42603103](https://stackoverflow.com/questions/42603103/how-to-completely-uninstall-vs-code-on-mac).
4. Deleted VS Code server from your remote server via:
- `sudo apt purge code` (deleted VS Code from your remote server)
- `rm -rf .vscode-server` (delete VS Code server folder from your `$HOME` directory)
- `sudo apt autoremove` (delete everything that have not been previously deleted including all dependencies of VS code)
!!**NOTE**!! use this command carefully as it will erase all dependencies that have not been deleted previously from your system.
- _`rm -rf .ipython`_ (perhaps this command is not required, but I executed it in the pipeline) 
5. Download version of [July 2023 (version 1.81)](https://code.visualstudio.com/updates/v1_81) on your local machine.
6. !!**Disable**!! as VS code make autoupdates in background #[50162685](https://stackoverflow.com/questions/50162685/how-do-i-disable-vs-code-of-updating-itself).
7. Configure clean VS code as new one (copy `settings.json`, install required extensions, etc.)
8. Establish remote connection with your remote server via ssh from VS Code toolkit.
9. Install all extensions that are required on your remote server.

Again, this solution worked for me, but I cannot assure that it will work for all mac users.

**Comment by ductai199x** [comment]
> Discovered the solution that worked for me (**local**: macOS Ventura 13.6, M1, **remote server**: Ubuntu 22.04.3 LTS x86_64).
> 
> 1. Backup `settings.json` (if you've made any local configurations).
> 2. Close VS Code locally and remotely.
> 3. Completely delete VS Code from your local machine via steps in #[42603103](https://stackoverflow.com/questions/42603103/how-to-completely-uninstall-vs-code-on-mac).
> 4. Deleted VS Code server from your remote server via:
> - `sudo apt purge code` (deleted VS Code from your remote server)
> - `rm -rf .vscode-server` (delete VS Code server folder from your `$HOME` directory)
> - `sudo apt autoremove` (delete everything that have not been previously deleted including all dependencies of VS code)
> !!**NOTE**!! use this command carefully as it will erase all dependencies that have not been deleted previously from your system.
> - _`rm -rf .ipython`_ (perhaps this command is not required, but I executed it in the pipeline) 
> 5. Download version of [July 2023 (version 1.81)](https://code.visualstudio.com/updates/v1_81) on your local machine.
> 6. !!**Disable**!! updates as VS code make autoupdates in background #[50162685](https://stackoverflow.com/questions/50162685/how-do-i-disable-vs-code-of-updating-itself).
> 7. Configure clean VS code as new one (copy `settings.json`, install required extensions, etc.)
> 8. Establish remote connection with your remote server via ssh from VS Code toolkit.
> 9. Install all extensions that are required on your remote server.
> 
> Again, this solution worked for me, but I cannot assure that it will work for all mac users.

Yep, similar to my solution on windows or linux above. The important thing is to delete the .vscode_server folder from remote and uninstall local 1.83.1 vscode, then downgrade to 1.83.0 and reinstall everything on remote. It sucked because I have to use about 10 remote servers daily.

**Comment by jzyrobert** [comment]
This is also happening on version 1.82.2 on a remote connection to a Ubuntu 22.04 host.

- Open ipynb with extensions disabled -> fine
- Open ipynb with extensions enabled -> Crashes right after activating extensions, before even selecting kernel
- Have tried deleting all extensions and reinstalling them, worked shortly at one point but then started crashing again

**Comment by rebornix** [comment]
We figured out it was caused by Pylance extension killing the remote extension host process (see details in https://github.com/microsoft/pylance-release/issues/4983). The Pylance team has fixed this issue in Prerelease and it working on a recovery release for stable. To mitigate/workaround this issue before the fix lands:

* Open an empty Remote SSH window (Remote indicator -> Connect to Host ... -> Pick the remote)
* Downgrade the Pylance extension to previous version (a week ago) or disable it for now

We will keep you posted once there is a new Pylance release.
